subsys/zigbee/cli/zigbee_cli_cmd_bdb.c

```diff
@@ -0,0 +1,864 @@
+
+	if (parse_hex_str(argv[1], strlen(argv[1]), key, sizeof(key), false)) {
+		zb_secur_setup_nwk_key(key, 0);
+	} else {
+		zb_cli_print_error(shell, "Failed to parse key", ZB_FALSE);
+		return -EINVAL;
+	}
+
+	zb_cli_print_done(shell, ZB_FALSE);
+	return 0;
+}
+
+/**@brief Perform a factory reset via local action
+ *
+ * See Base Device Behavior specification chapter 9.5 for details.
+ *
+ * @code
+ * > bdb factory_reset
+ * Done
+ * @endcode
+ */
+static int cmd_zb_factory_reset(const struct shell *shell, size_t argc,
+				char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	ZB_SCHEDULE_APP_CALLBACK(zb_bdb_reset_via_local_action, 0);
+	zb_cli_print_done(shell, ZB_FALSE);
+	return 0;
+}
+
+/**@brief Set amount of the child devices which can be connected to the device.
+ *
+ * @code
+ * bdb child_max <d:children_nbr>
+ * @endcode
+ *
+ *  @pre Setting only before @ref start "bdb start".
+ *
+ * If the argument is provided and `children_nbr` is in [0:32] range,
+ * set to that number. Otherwise, return error.
+ *
+ *
+ * Example:
+ * @code
+ * > bdb child_max 16
+ * Setting max children to: 16
+ * Done
+ * @endcode
+ */
+#ifndef ZB_ED_ROLE
+static int cmd_child_max(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+
+	uint32_t child_max = 0xFFFFFFFF;
+
+	/* Two argc - set the amount of the max_children. */
+	if (zigbee_is_stack_started()) {
+		zb_cli_print_error(shell, "Stack already started", ZB_FALSE);
+		return -ENOEXEC;
+	}
+
+	zb_cli_sscan_uint(argv[1], (uint8_t *)&child_max, 4, 10);
+	if (child_max > 32) {
+		zb_cli_print_error(shell, "Children device number must be within [0:32]",
+				   ZB_FALSE);
+		return -EINVAL;
+	} else {
+		/* Set the value by calling ZBOSS API. */
+		zb_set_max_children(child_max);
+		shell_print(shell, "Setting max children to: %d", child_max);
+	}
+
+	zb_cli_print_done(shell, ZB_FALSE);
+	return 0;
+}
+#endif
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ic,
+	SHELL_CMD_ARG(add, NULL, IC_ADD_HELP, cmd_zb_install_code, 3, 0),
+#ifndef ZB_ED_ROLE
+	SHELL_CMD_ARG(policy, NULL, IC_POLICY_HELP, cmd_zb_install_code, 2, 0),
+	SHELL_CMD_ARG(set, NULL, IC_SET_POLICY, cmd_zb_install_code, 2, 0),
+#endif
+	SHELL_SUBCMD_SET_END);
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_legacy,
+	SHELL_CMD_ARG(disable, NULL, "Disable legacy mode.",
+		      cmd_zb_legacy_disable, 1, 0),
+	SHELL_CMD_ARG(enable, NULL, "Enable legacy mode.",
+		      cmd_zb_legacy_enable, 1, 0),
+	SHELL_SUBCMD_SET_END);
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_role,
+	SHELL_CMD_ARG(zc, NULL, "Set Coordinator role.",
+		      cmd_zb_role_zc, 1, 0),
+	SHELL_CMD_ARG(zed, NULL, "Set End Device role.",
+		      cmd_zb_role_zed, 1, 0),
+	SHELL_CMD_ARG(zr, NULL, "Set Router role.",
+		      cmd_zb_role_zr, 1, 0),
+	SHELL_SUBCMD_SET_END);
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_bdb,
+	SHELL_CMD_ARG(channel, NULL, CHANNEL_HELP, cmd_zb_channel, 1, 1),
+#ifndef ZB_ED_ROLE
+	SHELL_CMD_ARG(child_max, NULL, CHILD_MAX_HELP, cmd_child_max, 2, 0),
```

Comment:
This could look like that:
`SHELL_COND_CMD(ZB_ED_ROLE, child_max, NULL, CHILD_MAX_HELP, cmd_child_max, 2, 0),`